### PR TITLE
Store feature toggles in session storage

### PR DESCRIPTION
**Pull request by theodur**
## Description
Stores feature toggles in session storage to reduce API calls.

As a result of storing feature toggles in session storage, some Cypress tests failed due to expecting a mock call to the feature toggle endpoint. Enabling the experimental feature [experimentalSessionAndOrigin](https://docs.cypress.io/api/commands/session#Syntax) resolved most of the test failures. This feature clears the page before each test along with all active session data. This behavior will be the default in a future major Cypress update. A few test suites needed to be refactored to visit the URL before each test in order to work with the experimental feature turned on.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#48374

## Testing done
Tested locally and in CI.

## Acceptance criteria
- [x] Feature toggle values are stored in sessionStorage once fetched.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/coronavirus-research/sign-up/tests/reviewAndSubmit.cypress.spec.js [hunk 1]
```diff
@@ -3,16 +3,14 @@ import featureTogglesEnabled from './fixtures/toggle-covid-feature.json';
 
 describe('COVID-19 Research Form', () => {
   describe('when entering valid information and submitting the sign up form', () => {
-    before(() => {
+    beforeEach(() => {
```
Comment (theodur) [inline src/applications/coronavirus-research/sign-up/tests/reviewAndSubmit.cypress.spec.js:6]: With `experimentalSessionAndOrigin` enabled, session data is reset before each test, so in order to get this test passing, the URL needs to be visited before each test, along with the feature toggle call being mocked

File: src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js [hunk 2]
```diff
@@ -14,11 +14,11 @@ describe(manifest.appName, () => {
       },
     };
     cy.intercept('GET', '/v0/feature_toggles*', featureToggles);
+    cy.visit(manifest.rootUrl);
```
Comment (theodur) [inline src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js:17]: This needed to be updated for a [similar reason](https://github.com/department-of-veterans-affairs/vets-website/pull/22455/files#r1004005542)

**Review by it-harrison — APPROVED** [review]

File: config/cypress.config.js [hunk 3]
```diff
@@ -192,6 +192,7 @@ const cypressConfig = {
     baseUrl: 'http://localhost:3001',
     specPattern: 'src/**/tests/**/*.cypress.spec.js?(x)',
     supportFile: 'src/platform/testing/e2e/cypress/support/index.js',
+    experimentalSessionAndOrigin: true,
```
Comment (alexandec) [inline config/cypress.config.js:195]: I'm definitely in favor of clearing session storage before every test. It did occur to me, we could run this by QA people to make sure they're aware of the change

**Review by alexandec — APPROVED** [review]

**Review by CBonade — APPROVED** [review]

File: src/applications/coronavirus-research/sign-up/tests/reviewAndSubmit.cypress.spec.js (hunk 1 above)
Comment (pjhill) [reply src/applications/coronavirus-research/sign-up/tests/reviewAndSubmit.cypress.spec.js:6]: Looks like a good change at any rate.

**Review by pjhill — APPROVED** [review]
LGTM

**Review by stephenBarrs — APPROVED** [review]
LGTM!

**Review by reblg — APPROVED** [review]

**Review by yonidoronpeters — APPROVED** [review]

**Review by alexandec — APPROVED** [review]

**Review by ericboehs — APPROVED** [review]

**Pull request merged**